src/users.ts

```diff
@@ -0,0 +1,901 @@
+import Errors from './errors'
+
+export interface User {
+  name?: string
+  id?: string
+  credentials?: {
+    identity: string
+    secret?: string
+  }
+  owner?: string
+  tags?: [string, string]
+  role?: string
+  status?: 'enabled' | 'disabled'
+  createdAt?: string
+  updatedAt?: string
+  updatedBy?: string
+}
+
+interface PageRes {
+  total: number
+  offset: number
+  limit: number
+}
+
+interface UsersInterface {
+  users: User[]
+  page: PageRes
+}
+
+interface Group {
+  id?: string
+  name?: string
+  status?: 'enabled' | 'disabled'
+  createdAt?: string
+  updatedAt?: string
+}
+interface Groups {
+  groups: Group[]
+  page: PageRes
+}
+
+interface Thing {
+  id?: string
+  name?: string
+  status?: 'enabled' | 'disabled'
+  createdAt?: string
+  updatedAt?: string
+}
+
+interface Things {
+  things: Thing[]
+  page: PageRes
+}
+
+interface Channel {
+  id?: string
+  name?: string
+  status?: 'enabled' | 'disabled'
+  createdAt?: string
+  updatedAt?: string
+}
+
+interface Channels {
+  channel: Channel[]
+  page: PageRes
+}
+
+interface Login {
+  identity?: string
+  secret?: string
+  domain_id?: string
+}
+
+interface QueryParams {
```

Comment:
These are not the only query params available. Check the `pageMetadata` in go-sdk and add those as the query params